### PR TITLE
hal: esp32c6: Clock control for USB JTAG peripheral

### PR DESCRIPTION
**Pull request by raffarost**
Added LL files for clock control of USB JTAG peripheral

### CODE REVIEW

**Review by sylvioalves — APPROVED** [review]

File: components/hal/esp32c6/include/hal/usb_serial_jtag_ll.h [hunk 1]
```diff
@@ -1,23 +1,22 @@
...
+
+#include <stdbool.h>
+#include "esp_attr.h"
+#include "soc/pcr_struct.h"
 #include "soc/usb_serial_jtag_reg.h"
 #include "soc/usb_serial_jtag_struct.h"
+#include "hal/usb_serial_jtag_types.h"
 
-#ifdef __cplusplus
-extern "C" {
```
Comment (marekmatej) [inline components/hal/esp32c6/include/hal/usb_serial_jtag_ll.h:14]: Any reason for removing the CPP ifdef?

File: components/hal/esp32c6/include/hal/usb_serial_jtag_ll.h [hunk 2]
```diff
@@ -168,6 +180,154 @@ static inline void usb_serial_jtag_ll_txfifo_flush(void)
     USB_SERIAL_JTAG.ep1_conf.wr_done=1;
 }
 
+/**
+ * @brief Enable USJ JTAG bridge
```
Comment (marekmatej) [inline components/hal/esp32c6/include/hal/usb_serial_jtag_ll.h:184]: What is USJ??

**Review by marekmatej — APPROVED** [review]

File: components/hal/esp32c6/include/hal/usb_serial_jtag_ll.h (hunk 1 above)
Comment (raffarost) [reply components/hal/esp32c6/include/hal/usb_serial_jtag_ll.h:14]: This file is original from IDF as is

File: components/hal/esp32c6/include/hal/usb_serial_jtag_ll.h (hunk 2 above)
Comment (raffarost) [reply components/hal/esp32c6/include/hal/usb_serial_jtag_ll.h:184]: I'll have a look on it to review

**Review by wmrsouza — APPROVED** [review]

**Review by LucasTambor — APPROVED** [review]

File: components/hal/esp32c6/include/hal/usb_serial_jtag_ll.h (hunk 2 above)
Comment (sylvioalves) [reply components/hal/esp32c6/include/hal/usb_serial_jtag_ll.h:184]: Will you rename it to USB?

Comment (raffarost) [reply components/hal/esp32c6/include/hal/usb_serial_jtag_ll.h:184]: Yes, I'll do a quick review on the files now and update

File: components/hal/esp32c6/include/hal/usb_serial_jtag_ll.h (hunk 1 above)
Comment (raffarost) [reply components/hal/esp32c6/include/hal/usb_serial_jtag_ll.h:14]: I see this block was just moved to stay right before the functions definition, so it's still there

File: components/hal/esp32c6/include/hal/usb_serial_jtag_ll.h (hunk 2 above)
Comment (raffarost) [reply components/hal/esp32c6/include/hal/usb_serial_jtag_ll.h:184]: It appears we have several references to USJ as the USB JTAG module (might be USb Jtag?), nine in this file alone and also present in 'usb_serial_jtag_types.h'. I'm not sure it is the case to change it.

**Pull request merged**